### ISSUE TITLE
Crash when using left swipe backspace in termux

### ISSUE THREAD

**Issue by FuturisticGoo** [florisboard/florisboard#967]
#### Short description
Florisboard crashes when trying to use left swipe backspace in [termux](https://f-droid.org/en/packages/com.termux/). The crash is reproducible. Normal backspace works fine. Left swipe backspace works in other text inputs as well, only crashes in termux

#### Steps to reproduce
1. Go to Termux
2. Write something (or don't, crashes either way)
3. Try to do left swipe backspace
4. See crash

#### Screen recording
Recording shows it working on other apps as expected and then crashing in termux
 
https://user-images.githubusercontent.com/28504180/120176422-e3b78b00-c224-11eb-9ccb-15435c44a621.mp4

#### Environment information
- FlorisBoard Version: 0.3.13-beta02 (45)
- Device: Samsung SM-J700F
- Android: 10 (cn=Q sdk=29)

#### Features enabled
```kt
smartbar = true
suggestions = true
suggestions_clipboard = true
suggestions_next_word = true
glide = true
clipboard_internal = true
clipboard_history = true
```

#### Attached stacktrace files
<details>
<summary>1622453674354.stacktrace</summary>

```
java.lang.IllegalArgumentException: Cannot coerce value to an empty range: maximum -1 is less than minimum 0.
	at kotlin.ranges.RangesKt___RangesKt.coerceIn(_Ranges.kt:1103)
	at dev.patrickgold.florisboard.ime.text.keyboard.TextKeyboardView.handleDeleteSwipe(TextKeyboardView.kt:588)
	at dev.patrickgold.florisboard.ime.text.keyboard.TextKeyboardView.onSwipe(TextKeyboardView.kt:535)
	at dev.patrickgold.florisboard.ime.text.gestures.SwipeGesture$Detector.onTouchMove(SwipeGesture.kt:136)
	at dev.patrickgold.florisboard.ime.text.keyboard.TextKeyboardView.onTouchEventInternal(TextKeyboardView.kt:295)
	at dev.patrickgold.florisboard.ime.keyboard.KeyboardView$1.invokeSuspend(KeyboardView.kt:52)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:227)
	at kotlinx.coroutines.DispatchedTaskKt.resumeUnconfined(DispatchedTask.kt:190)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:161)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:362)
	at kotlinx.coroutines.CancellableContinuationImpl.completeResume(CancellableContinuationImpl.kt:479)
	at kotlinx.coroutines.channels.AbstractChannel$ReceiveHasNext.completeResumeReceive(AbstractChannel.kt:940)
	at kotlinx.coroutines.channels.ArrayChannel.offerInternal(ArrayChannel.kt:84)
	at kotlinx.coroutines.channels.AbstractSendChannel.offer(AbstractChannel.kt:141)
	at kotlinx.coroutines.channels.ChannelsKt__ChannelsKt.sendBlocking(Channels.kt:21)
	at kotlinx.coroutines.channels.ChannelsKt.sendBlocking(Unknown Source:1)
	at dev.patrickgold.florisboard.ime.keyboard.KeyboardView.onTouchEvent(KeyboardView.kt:79)
	at android.view.View.dispatchTouchEvent(View.java:13953)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:465)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1849)
	at android.app.Dialog.dispatchTouchEvent(Dialog.java:861)
	at android.inputmethodservice.SoftInputWindow.dispatchTouchEvent(SoftInputWindow.java:156)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:423)
	at android.view.View.dispatchPointerEvent(View.java:14212)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5649)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5452)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4955)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5008)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4974)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5114)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4982)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5171)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4955)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5008)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4974)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4982)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4955)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7672)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7641)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7602)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7797)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:188)
	at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
	at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:178)
	at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:7748)
	at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:7821)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
	at android.view.Choreographer.doCallbacks(Choreographer.java:791)
	at android.view.Choreographer.doFrame(Choreographer.java:719)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

```
</details>

**Comment by patrickgold** [comment]
Thanks for your bug report! The crash itself makes sense, because you cannot select text in a terminal app with the standard IME API of Android. Though normally there's a safeguard preventing this crash for terminal apps within FlorisBoard, which seems to fail for some reason. Will investigate into this!

**Comment by patrickgold** [comment]
Fixed in #970, will be released later today in beta03.
